server.js

```diff
@@ -24,6 +27,310 @@ app.get("/", (req, res) => {
+  const ratingForShow = netflixTitlesData.filter((item) => {
+    return item.rating.toLowerCase().includes(rating.toLowerCase());
+  });
+  if (ratingForShow.length) {
+    res.status(200).json({
+      success: true,
+      message: "Titles found with the provided rating",
+      body: {
+        title: ratingForShow
+      }
+    });
+  } else {
+    res.status(404).json({
+      success: false,
+      message: "No titles found with the provided rating",
+      body: {}
+    });
+  }
+});
+
+//Search by duration
+app.get('/duration/:duration', (req, res) => {
+  const { duration } = req.params;
+  const durationForShow = netflixTitlesData.filter((item) => {
+    return item.duration.toLowerCase().includes(duration.toLowerCase());
+  });
+  if (durationForShow.length) {
+    res.status(200).json({
+      success: true,
+      message: "Titles found with the provided duration",
+      body: {
+        title: durationForShow
+      }
+    });
+  } else {
+    res.status(404).json({
+      success: false,
+      message: "No titles found with the provided duration",
+      body: {}
+    });
+  }
+});
+
+// Search by genre
+app.get('/listed_in/:listed_in', (req, res) => {
+  const { listed_in } = req.params;
+  const ListedInForShow = netflixTitlesData.filter((item) => {
+    return item.listed_in.toLowerCase().includes(listed_in.toLowerCase());
+  });
+  if (ListedInForShow.length) {
+    res.status(200).json({
+      success: true,
+      message: "Titles found with the provided listed in",
+      body: {
+        title: ListedInForShow
+      }
+    });
+  } else {
+    res.status(404).json({
+      success: false,
+      message: "No titles found with the provided listed in",
+      body: {}
+    });
+  }
+});
+
+//Search by description
+app.get('/description/:description', (req, res) => {
+  const { description } = req.params;
+  const descriptionForShow = netflixTitlesData.filter((item) => {
+    return item.description.toLowerCase().includes(description.toLowerCase());
+  });
+  if (descriptionForShow.length) {
+    res.status(200).json({
+      success: true,
+      message: "Titles found based on the description provided",
+      body: {
+        title: descriptionForShow
+      }
+    });
+  } else {
+    res.status(404).json({
+      success: false,
+      message: " No titles found based on the description provided",
+      body: {}
+    });
+  }
+});
+
+// Search by type
+app.get('/type/:type', (req, res) => {
+  const { type } = req.params;
+  const typeForShow = netflixTitlesData.filter((item) => {
+    return item.type.toLowerCase().includes(type.toLowerCase());
+  });
+  if (typeForShow.length) {
+    res.status(200).json({
+      success: true,
+      message: "Titles found based on the type provided",
+      body: {
+        title: typeForShow
+      }
+    });
+  } else {
+    res.status(404).json({
+      success: false,
+      message: "No titles found based on the type provided",
+      body: {}
+    });
+  }
+});
```

Comment:
You have done an impressive amount of endpoints. The title of endpoint Guru is yours to keep!